client/connector_remove.go

```diff
@@ -3,54 +3,39 @@ package client
 import (
 	"context"
 	"fmt"
-	"strings"
 
 	"k8s.io/client-go/util/retry"
 
 	"github.com/skupperproject/skupper/api/types"
 	"github.com/skupperproject/skupper/pkg/kube"
 	"github.com/skupperproject/skupper/pkg/qdr"
-	"github.com/skupperproject/skupper/pkg/utils/configs"
 )
 
-func removeConnector(name string, list []types.Connector) (bool, []types.Connector) {
-	updated := []types.Connector{}
-	found := false
-	for _, c := range list {
-		if c.Name != name {
-			updated = append(updated, c)
-		} else {
-			found = true
-		}
-	}
-	return found, updated
-}
-
 func (cli *VanClient) ConnectorRemove(ctx context.Context, options types.ConnectorRemoveOptions) error {
 	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		current, err := kube.GetDeployment(types.TransportDeploymentName, options.SkupperNamespace, cli.KubeClient)
+		deployment, err := kube.GetDeployment(types.TransportDeploymentName, options.SkupperNamespace, cli.KubeClient)
+		if err != nil {
+			return err
+		}
+		configmap, err := kube.GetConfigMap("skupper-internal", options.SkupperNamespace, cli.KubeClient)
+		if err != nil {
+			return err
+		}
+		current, err := qdr.GetRouterConfigFromConfigMap(configmap)
 		if err != nil {
 			return err
 		}
-		mode := qdr.GetTransportMode(current)
-		found, connectors := removeConnector(options.Name, qdr.ListRouterConnectors(mode, options.SkupperNamespace, cli.KubeClient))
+
+		found := current.RemoveConnector(options.Name)
```

Comment:
On a disconnect, I do not see connector removed from cm.

iiuc, need current.UpdateConfigMap(configmap) prior to update

Reply:
Well spotted, thanks!